display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java

```diff
@@ -0,0 +1,351 @@
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import org.junit.Before;
+import org.junit.Test;
+import org.testfx.framework.junit.ApplicationTest;
+import org.testfx.robot.Motion;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.fail;
+
+public class SketchOnMapTest extends ApplicationTest {
+
+  MapView mapView;
+  GraphicsOverlay graphicsOverlay;
+
+  Button editButton;
+  Button stopButton;
+
+  Button pointButton;
+  Button multipointButton;
+  Button polylineButton;
+  Button polygonButton;
+  Button freehandPolylineButton;
+  Button freehandPolygonButton;
+
+  Button undoButton;
+  Button redoButton;
+  Button saveButton;
+  Button clearButton;
+
+  @Override
+  public void start(Stage stage) throws Exception {
+    // set up the scene
+    FXMLLoader loader = new FXMLLoader(getClass().getResource("/sketch_on_map.fxml"));
+    Parent root = loader.load();
+    Scene scene = new Scene(root);
+
+    stage.setScene(scene);
+    stage.setTitle("Sketch on Map Sample");
+    stage.setWidth(1000);
+    stage.setHeight(700);
+    stage.show();
+
+    // wait for initialization
+    sleep(1500);
+  }
+
+  @Before
+  public void setup() {
+
+    // get a handle on the MapView and GraphicsOverlay
+    Node mapViewNode = lookup("#mapView").query();
+    if (mapViewNode instanceof MapView) {
+      mapView = (MapView) mapViewNode;
+      graphicsOverlay = mapView.getGraphicsOverlays().get(0);
+
+    } else {
+      fail("MapView Node could not be found");
+    }
+
+    // find the middle button group and get each button
+    VBox vbox = lookup(".panel-region").queryAs(VBox.class);
+    GridPane middleButtonGroup = (GridPane) vbox.getChildren().get(2);
+    pointButton = (Button) middleButtonGroup.getChildren().get(0);
+    multipointButton = (Button) middleButtonGroup.getChildren().get(1);
+    polylineButton = (Button) middleButtonGroup.getChildren().get(2);
+    polygonButton = (Button) middleButtonGroup.getChildren().get(3);
+    freehandPolylineButton = (Button) middleButtonGroup.getChildren().get(4);
+    freehandPolygonButton = (Button) middleButtonGroup.getChildren().get(5);
+
+    editButton = lookup("#editButton").queryButton();
+    stopButton = lookup("#stopButton").queryButton();
+    undoButton = lookup("#undoButton").queryButton();
+    redoButton = lookup("#redoButton").queryButton();
+    saveButton = lookup("#saveButton").queryButton();
+    clearButton = lookup("#clearButton").queryButton();
+  }
+
+  /**
+   * Test Case 1: Use the SketchEditor to draw a point on the map, and verify that it is added to the GraphicsOverlay.
+   */
+  @Test
+  public void drawPoint() {
+    clickOn(pointButton);
+    clickOn(mapView);
+    clickOn(saveButton);
+
+    ListenableList<Graphic> graphics = graphicsOverlay.getGraphics();
+    assertEquals(1, graphics.size());
+    assertEquals(GeometryType.POINT, graphics.get(0).getGeometry().getGeometryType());
```

Comment:
You might also want to verify the point is in the correct location

Reply:
Yep, added this. Would it be ok to leave this out of the remainder of the tests, seeing that if this passes it proves that SketchEditor translates the screen point to the map point correctly?